zone/entity.cpp

```diff
@@ -1863,21 +1863,56 @@ Client *EntityList::GetClientByLSID(uint32 iLSID)
 	return nullptr;
 }
 
+#ifdef BOTS
+Bot* EntityList::GetRandomBot(const glm::vec3& location, float distance, Bot* exclude_bot)
+{
+	auto is_whole_zone = false;
+	if (location.x == 0.0f && location.y == 0.0f) {
```

Comment:
This is kind of strange logic, why include whole zone when coordinates are 0 ?

Reply:
If you don’t supply a location, it assumes you want the entire zone. glm::vec3 is set to 0.0 for XYZ in header if not sent.